src/integrations/xiaomi-mi/xiaomi-mi.service.ts

```diff
@@ -0,0 +1,211 @@
+        this.entitiesService
+      );
+    });
+    noble.on('stateChange', XiaomiMiService.handleStateChange);
+    noble.on('discover', this.handleDiscovery.bind(this));
+    noble.on('warning', this.onWarning.bind(this));
+    this.logger.log('XiaomiMi Service bootstrapped!');
+  }
+
+  /**
+   * Log warnings from noble.
+   */
+  onWarning(message: string): void {
+    this.logger.warn('Warning: ', message);
+  }
+
+  /**
+   * Determines whether we have any sensors.
+   *
+   * @returns Sensors status
+   */
+  hasSensors(): boolean {
+    return this.config.sensors?.length > 0;
+  }
+
+  /**
+   * Filters found BLE peripherals and publishes new readings to sensors, depending on configuration.
+   *
+   * @param peripheral - BLE peripheral
+   */
+  handleDiscovery(peripheral: Peripheral): void {
+    const { advertisement, id } = peripheral || {};
+    let sensorSet = this.sensors[id];
+    if (!sensorSet) {
+      return;
+    }
+    const buffer = this.getValidServiceData(advertisement);
+    if (!buffer) {
+      return;
+    }
+    const serviceData = this.parseServiceData(buffer);
+    if (!serviceData) {
+      return;
+    }
+    const event = serviceData.event;
+    switch (serviceData.eventType) {
+      case EventTypes.temperature: {
+        sensorSet.recordMeasure('temperature', '°C', event.temperature);
+        break;
+      }
+      case EventTypes.humidity: {
+        sensorSet.recordMeasure('humidity', '%', event.humidity);
+        break;
+      }
+      case EventTypes.battery: {
+        this.logger.log(`Got battery: ${event.battery}`);
+        sensorSet.recordMeasure('battery', '%', event.battery);
+        break;
+      }
+      case EventTypes.temperatureAndHumidity: {
+        this.logger.log(
+          `Got temperature: ${event.temperature} and humidity: ${event.humidity}`
+        );
+        sensorSet.recordMeasure('temperature', '°C', event.temperature);
+        sensorSet.recordMeasure('humidity', '%', event.humidity);
+        break;
+      }
+      case EventTypes.illuminance: {
+        this.logger.log(`Got illuminance: ${event.illuminance}`);
+        break;
+      }
+      case EventTypes.moisture: {
+        this.logger.log(`Got moisture: ${event.moisture}`);
+        break;
+      }
+      case EventTypes.fertility: {
+        this.logger.log(`Got fertility: ${event.fertility}`);
+        break;
+      }
+      default: {
+        this.logger.error(`Unknown event type: ${serviceData.eventType}`);
+        break;
+      }
+    }
+  }
+
+  getValidServiceData(advertisement: Advertisement): Buffer {
+    if (!advertisement || !advertisement.serviceData) {
+      return null;
+    }
+    const uuidPair = advertisement.serviceData.find(
+      data => data.uuid.toLowerCase() === SERVICE_DATA_UUID
+    );
+    if (!uuidPair) {
+      return null;
+    }
+    return uuidPair.data;
+  }
+
+  parseServiceData(buffer: Buffer): ServiceData | null {
+    try {
+      return new Parser(buffer, null).parse();
```

Comment:
There is no way to configure a `bindKey` - if there would be an additional nullable option per device that could then be passed to the Parser here we could potentially add support for more devices without much extra effort right?

Reply:
Fixed.